reclist/metrics/standard_metrics.py

```diff
@@ -127,4 +133,62 @@ def precision_at_k(y_preds, y_test, k=3):
 
 def recall_at_k(y_preds, y_test, k=3):
     recall_ls = [len(set(_y).intersection(set(_p[:k]))) / len(_y) if _y else 1 for _p, _y in zip(y_preds, y_test)]
-    return np.average(recall_ls)
\ No newline at end of file
+    return np.average(recall_ls)
+
+
+def rec_items_distribution_at_k(y_preds, k=3, bin_width=100, debug=True):
+    def _calculate_hist(frequencies, bins):
+        """ Works out the counts of items in each bucket """
+        counts_per_bin = Counter([bisect_right(bins, item) - 1 for item in frequencies])
+        counts_per_bin_list = list(counts_per_bin.items())
+        empty_bins_indices = [ele for ele in np.arange(len(bins) - 1) if ele not in [
+            index for index, _ in counts_per_bin_list
+        ]]
+        counts_per_bin_list.extend([(index, 0) for index in empty_bins_indices])
+        counts_per_bin_sorted = sorted(counts_per_bin_list, key=lambda x: x[0], reverse=False)
+        return [y for _, y in counts_per_bin_sorted]
+
+    def _format_results(bins, counts_per_bin):
+        """ Formatting results """
+        results = {}
+        for index, (_, count) in enumerate(zip(bins, counts_per_bin)):
+            if bins[index] != bins[index + 1]:
+                results[str(bins[index]) + '-' + str(bins[index + 1] - 1)] = count
+        return results
+
+    # estimate frequency of recommended items in predictions
+    reduce_at_k_preds = [preds[:k] for preds in y_preds]
+    counts = Counter(chain.from_iterable(reduce_at_k_preds))
+
+    frequencies = list(counts.values())
+
+    # fixed bin size
+    bins = np.arange(np.min(frequencies), np.max(frequencies) + bin_width, bin_width)
+    counts_per_bin_sorted = _calculate_hist(frequencies, bins)
+
+    # log bin size
+    log_bins = np.logspace(np.log10(bins[0]), np.log10(bins[-1]), len(bins))
+    log_bins = np.array([int(round(log_bin, 2)) for log_bin in log_bins])
+    counts_per_logbin_sorted = _calculate_hist(frequencies, log_bins)
+
+    test_results = {
+        'histogram_fixed': _format_results(bins, counts_per_bin_sorted),
+        'histogram_log': _format_results(log_bins, counts_per_logbin_sorted)
+    }
+
+    if debug:
+        f, (ax1, ax2) = plt.subplots(2, 1)
+
+        # debug / visualization
+        ax1.bar(bins[:-1], counts_per_bin_sorted,
+                width=bin_width, align='edge')
+
+        ax1.set_ylabel('Number of items')
+        ax2.bar(log_bins[:-1], counts_per_logbin_sorted,
+                width=(log_bins[1:] - log_bins[:-1]), align='edge')
+        ax2.set_xscale('log', base=10)
+        ax2.set_xlabel('Frequency of being recommended')
```

Comment:
What about `No. of times an item appear in a recs list`? We can reduce the fontsize, if needed.